docs/SECURITY.md

```diff
@@ -0,0 +1,81 @@
+
+## Security capabilities vs. security policies
+
+At the end of the day, EVE only does what its controller instructs it to do. This puts an enormous premium on protecting the trust between EVE instance and its controller at all costs (in both direction) but it also allows us to have a clear division of responsibilities between the two:
+
+* EVE only implements security capabilities, never security policies
+* Controller orchestrates EVE's security capabilities in order to realize sophisticated policies
+
+On the EVE's side, this separation allows us to focus on a small number of well designed capabilities (AKA security building blocks) that are guaranteed to have a well defined interactions between each other. For example, while EVE may provide a capability for storing encrypted binary blobs on the Edge Nodes it will be up to the controller's security policy to determine which pieces of data will be stored there.
+
+Controller and its management of security policies are out of scope for this document.
+
+## EVE threat model
+
+Borrowing a great deal from Chromium effort we consider two different kinds of adversaries:
+
+* An opportunistic adversary
+* A dedicated adversary
+
+The opportunistic adversary is just trying to compromise an individual Edge Node and/or data.  They are not targeting a specific user or enterprise, and they are not going to steal, disassemble or modify the Edge Node but they are very likely to have prolonged physical access to it. This level of physical access allows opportunistic adversary a chance to replace trusted network and I/O connections with those of their own making, though. Unlike Chromium, we assume that this will facilitate things like DNS or other network-level attacks and will place them within an opportunistic adversary's reach.
+
+The dedicated adversary may target a user or an enterprise specifically for attack.  They are willing to steal Edge Nodes to recover data or account credentials (not just to re-sell the device to make money). They are also willing and capable of modifying an Edge Node with extra hardware and software components. They may also do anything that the opportunistic adversary can do.
+
+For now we are mainly focusing on dangers posed by opportunistic adversaries.
+
+## Establishing trust between EVE and EVE's controller
+
+### EVE trusting its controller
+
+Recall that EVE's deployment model presupposes a controller that can exercise arbitrary control over Edge Nodes. EVE provides the following capabilities that can protect against an adversary trying to take control over an Edge Node by pretending to be a controller:
+
+* The controller's network address (hostname and port) is considered immutable and can only be changed by a total reinstall of EVE
```

Comment:
Does this conflict with the very premise of open source EVE? Specifically, that replacing the controller does _not_ require a complete truck roll to every single node?

Reply:
Yes it does. And that's exactly why I'm calling it out. Do you propose that we also highlight it like that by saying "this is one of the very, very few places where we specifically contradict our own principles of never doing a truck roll"?

Reply:
Yes, if we are violating our own principles, we should highlight it here, although perhaps with better language than "contradict/violate our own principles."

The more fundamental issue here is that there are several key reasons we are doing OSS> One of them is that companies can feel comfortable signing up with Zededa/ControllerX/ControllerY, because if Z/X/Y goes out of business, then it doesn't take a full truck roll to visit every single device to switch Controllers."

It isn't just that it violates our principles of what we are trying to do _while you have EVE with Z/X/Y_, it is that it may upend one of the very purposes of having EVE be standard.

We may need to continue this conversation in Slack, but I think this actually is fundamental.

Reply:
Please take a look at the new language. There's absolutely no way we will be tackling this in the next year or so -- so what is important for now is to state the limitation clearly up-front while keeping the door open for the major EVE updates.

Reply:
We should move this to a real conversation, try and schedule some time to discuss. I really do think this is fundamental. If we cannot tackle It for a year, we need to have a clear path as to _how_ we will tackle it and when (the plan), and state that up front, or it will slip in the face of other pressures. It isn't that I don't believe us, it is just realism. 

Look at it this way: if switching from controller A to controller B requires a complete reinstall, then it doesn't have to be EVE on the next (or the first) install, it could be anything. The very premise of "EVE is _the_ standard that interoperates with any controller" largely is undercut. 

I get that we need some security ringfencing here to prevent a controller switch mid-flight, as that would be a strong attack vector. But making it explicitly _not_ updatable undermines the very premise of EVE as open standard.

Reply:
@deitch I suggest you open up a conversation over on eve-tsc@ and also use TSC meeting this coming Thu to talk about (sadly I won't be able to participate).

As far as doc goes, though, I think it now documents the current behvaiour and intent and seems like there's not much else to add here -- so for now I consider the matter closes.